ws-server/src/tests.rs

```diff
@@ -642,7 +643,7 @@ async fn custom_subscription_id_works() {
 
 	init_logger();
 	let server = WsServerBuilder::default()
-		.set_id_provider(HardcodedSubscriptionId)
+		.set_id_provider(Box::new(HardcodedSubscriptionId))
```

Comment:
Just curious really; what's the motivation for exposing the `Box`-ing here, rather than boxing it internally?

Reply:
the motivation/rationale is that if you have something like:

`Option<Box<IdProvider>>` then having `impl IdProvider` won't work, thus I changed it to `Box<dyn IdProvider>`

Reply:
Oh, I just meant that instead of `.set_id_provider(Box::new(HardcodedSubscriptionId))`, we could keep `.set_id_provider(HardcodedSubscriptionId)` and do the Boxing inside the function?

Reply:
I changed it back with your comment to implement `IdProvider for Box<T>`, so all good now anyway.

Reply:
Late to the conversation here, apologies. So the point of the test is now to prove that one *can* use a `Box<T>` (where `T: IdProvider`) if desired? It would be good imo to have a test that shows both ways, boxed and plain, and perhaps a comment that motivates why one might want to use the longer boxed version? And/or some docs on [the impl for `Box<T>`](https://github.com/paritytech/jsonrpsee/blob/1e00d8b2331bf34b46c704577505b0c4781f59eb/core/src/traits.rs#L79)?